base/workflow/R/run.write.configs.R

```diff
@@ -34,68 +35,82 @@ run.write.configs <- function(settings, write = TRUE, ens.sample.method = "unifo
+  if(is.logical(use.existing.samples)){
+    if(use.existing.samples & file.exists(file.path(settings$outdir, "samples.Rdata"))){
+      load(file.path(settings$outdir, "samples.Rdata")) ## loads ensemble.samples, trait.samples, sa.samples, runs.samples, env.samples
+    } else {
+      for (i in seq_along(settings$pfts)) {
+        ## if posterior.files is specified us that
+        if (is.na(posterior.files[i])) {
+          ## otherwise, check to see if posteriorid exists
+          if (!is.null(settings$pfts[[i]]$posteriorid)) {
+            #TODO: sometimes `files` is a 0x0 tibble and other operations with it fail.
+            files <- PEcAn.DB::dbfile.check("Posterior",
+                                            settings$pfts[[i]]$posteriorid, 
+                                            con, settings$host$name, return.all = TRUE)
+            pid <- grep("post.distns.*Rdata", files$file_name)  ## is there a posterior file?
+            if (length(pid) == 0) {
+              pid <- grep("prior.distns.Rdata", files$file_name)  ## is there a prior file?
+            }
+            if (length(pid) > 0) {
+              posterior.files[i] <- file.path(files$file_path[pid], files$file_name[pid])
+            }  ## otherwise leave posteriors as NA
+          }
+          ## otherwise leave NA and get.parameter.samples will look for local
+        } else {
+          ## does posterior.files point to a directory instead of a file?
+          if(utils::file_test("-d",posterior.files[i])){
+            pfiles = dir(posterior.files[i],pattern = "post.distns.*Rdata",full.names = TRUE)
+            if(length(pfiles)>1){
+              pid = grep("post.distns.Rdata",pfiles)
+              if(length(pid > 0)){
+                pfiles = pfiles[grep("post.distns.Rdata",pfiles)]
+              } else {
+                PEcAn.logger::logger.error(
+                  "run.write.configs: could uniquely identify posterior files within",
+                  posterior.files[i])
+              }
+              posterior.files[i] = pfiles
+            }
+          }
+          ## also, double check PFT outdir exists
+          if (is.null(settings$pfts[[i]]$outdir) || is.na(settings$pfts[[i]]$outdir)) {
+            ## no outdir
+            settings$pfts[[i]]$outdir <- file.path(settings$outdir, "pfts", settings$pfts[[i]]$name)
+          }
+        }  ## end else
+      } ## end for loop over pfts
+      
+      ## Sample parameters
+      PEcAn.uncertainty::get.parameter.samples(settings, posterior.files, ens.sample.method)
+      load(file.path(settings$outdir, "samples.Rdata")) ## loads ensemble.samples, trait.samples, sa.samples, runs.samples, env.samples
+      
+      ## end use.existing.samples == FALSE
+    } ## is.logical(use.existing.samples)
   } else {
-    PEcAn.logger::logger.error(samples.file, "not found, this file is required by the run.write.configs function")
+    ### use.existing.samples must be a filename
+    load(use.existing.samples)
```

Comment:
I did a doubletake here ("wait, why is it calling load on a bool?") even after reading at the top of the function that `use.existing.samples` could be a filepath.

Possible alternate design: Call this param `existing.sample.file`, have the argument always be a path, and make the default value be NULL (behaving like FALSE does now). Then passing `existing.sample.file = "samples.Rdata"` would behave like TRUE does now with no need for a special case.